### DISCUSSION TITLE
Default settings ImageMagick and different quality

### DISCUSSION THREAD

**Discussion by fastweb-grupp** [ImageMagick/ImageMagick#5141] [Consulting]
Hello,

Using online tools https://magickstudio.imagemagick.org/scripts/MagickStudio.cgi i upload jpg file, make resize and get good result of image quality, but if use command line in unix /usr/local/bin/convert 36826.jpg -resize 360x240 -quality 100 36826-small.jpg i get different relult, some details of the image have dim colors.

i use ImageMagick 7.1, screenshot results there https://drive.google.com/file/d/1hrtaBg6LheYycm6I-8bL8VZASZPwf-io/view?usp=sharing

![quality](https://user-images.githubusercontent.com/43278267/167464595-00771e63-86c7-470d-a011-bc3a3f9ae816.png)

maybe https://magickstudio.imagemagick.org/scripts/MagickStudio.cgi use other command?

Thanks

**Comment by fastweb-grupp** [comment]
original file there https://drive.google.com/file/d/1TOVQ8mNMIjH_yyjGOE3fieR8GxNPitpk/view?usp=sharing

![36826](https://user-images.githubusercontent.com/43278267/167464897-314abe76-f733-450d-a330-03b0d5fb354b.jpg)

**Comment by fmw42** [comment]
I do not see any issues. Be sure to set your resize filter to Lanczos in MagickStudio, since that is what will be used in command line as the default when your downsize.

When I repeat your commands (IM 6.9.12-48 Q16 and also IM 7.1.0-33 Q16-HDRI), I get:

Command Line:

![36826-small](https://user-images.githubusercontent.com/13458218/167489503-ac99423a-cb0c-4822-ba16-359f73da9b51.jpg)



MagickStudio:

![36826_small_studio](https://user-images.githubusercontent.com/13458218/167489524-a83325d3-fd8b-4fd5-947b-12e8e563919a.jpg)


`compare -metric rmse 36826-small.jpg 36826_small_studio.jpg null:`
58.2805 (0.000889303)

This is 0.09% difference (less than one tenth of percent)

**Comment by Jmuccigr** [comment]
> /usr/local/bin/convert 36826.jpg -resize 360x240 -quality 100 36826-small.jpg

I think @fmw42 controlled for this, but using the `convert` command doesn't do the same thing as `magick` in v. 7, which I assume is what the cgi is doing, does it?

**Reply by fmw42** [reply]
I do not understand your comment. Your command should work exactly the same in IM 7 as IM 6 apart from HDRI vs non-HDRI precision issues. Can you provide and example where they do not produce the same result? Indeed, one should use magick in IM 7, not convert and not magick convert to have proper IM 7 syntax and features.

**Reply by Jmuccigr** [reply]
> I do not understand your comment. Your command should work exactly the same in IM 7 as IM 6 apart from HDRI vs non-HDRI precision issues. Can you provide and example where they do not produce the same result? Indeed, one should use magick in IM 7, not convert and not magick convert to have proper IM 7 syntax and features.

I don't have such an example. I was relying on claims like your last sentence here (that one should use `magick` not `convert` in IM 7) and assuming that results might be different in certain circumstances.

**Comment by fastweb-grupp** [comment]
Hello, 

i record video for this problem [https://youtu.be/eTFOpHN8ERA](https://youtu.be/eTFOpHN8ERA)

Thanks

**Comment by fastweb-grupp** [comment]
ssh image (white-yellow color)
![ssh-20079-small-2](https://user-images.githubusercontent.com/43278267/169367435-7b259a46-dfcf-4001-9f77-654593a4f1e5.jpg)

magickstudio image (deep yellow color)
![magickstudio](https://user-images.githubusercontent.com/43278267/169368162-f7a362f4-719d-4003-aa67-f4ef0e8030f0.jpg)


different color

**Reply by fmw42** [reply]
Please show your original images and **your Imagemagick command line** and corresponding output.

**Comment by fastweb-grupp** [comment]
original file
![20079](https://user-images.githubusercontent.com/43278267/169370570-c1f070cc-4e41-4a15-9558-29c902cf0133.jpg)

**Comment by fastweb-grupp** [comment]
/usr/local/bin/magick 20079.jpg -resize 268x150 -filter Lanczos -quality 100 20079-small-2.jpg

**Comment by snibgo** [comment]
If you want to "-resize" with any particular "-filter", you need to put "-filter" before the "-resize".

**Reply by fastweb-grupp** [reply]
without changes

**Comment by fmw42** [comment]
Your image has no explicit profile. Add a sRGB.icc profile to it before processing.  

But since you are reducing the size using -filter lanczos, you will get blending of neighboring pixel colors in the reduced image. When you zoom it back out you will see blocks of colors that are the blends from the neighbors used by lanczos. If you do not want that, then use -filter point -resize or use -sample.

Also as user snibgo said above, put the -filter ... before -resize. Imagemagick 7 wants setting before the operators that use the settings.

**Reply by fastweb-grupp** [reply]
/usr/local/bin/magick 20079.jpg -colorspace sRGB -filter Lanczos -resize 268x150 -quality 100 20079-small-3.jpg

without changes

**Reply by fmw42** [reply]
Use -profile sRGB.icc rather than -colorspace sRGB.  The profile did not make a difference for me. But best to make sure you have one so that the display devices will show it consistently.

**Comment by fastweb-grupp** [comment]
why is there one picture on the website https://magickstudio.imagemagick.org and a completely different color through the command line?
what command website https://magickstudio.imagemagick.org send for resize image?

**Reply by fmw42** [reply]
This statement/question is not very clear. Please explain further.

**Reply by fastweb-grupp** [reply]
result of work web site https://magickstudio.imagemagick.org different of work ssh 
result = different color

**Comment by snibgo** [comment]
I downloaded your large input file as "foundation.jpg".

I resized it with...
```
magick foundation.jpg -resize 268x150 r.png
```
... with IM v7.1.0-32 on Windows 8.1.

Then I resized it with https://magickstudio.imagemagick.org/ to the same dimension, and downloaded the image as "ms.png".

Then I compared the two images:
```
magick compare -metric RMSE r.png ms.png NULL:

0 (0)
```
The results are identical.

I don't know whether MagickStudio can tell us what command it generates and executes.

**Reply by fastweb-grupp** [reply]
attach the files here for comparison

**Reply by fastweb-grupp** [reply]
![DIFFERENT-COLOR](https://user-images.githubusercontent.com/43278267/169563910-67ee3277-cfa0-4567-aea4-f32570ac7583.jpg)

**Comment by fastweb-grupp** [comment]
i try in windows and nothing, problem different color

**Comment by fastweb-grupp** [comment]
look words FO is different color, deep yellow and white-yellow

![FO](https://user-images.githubusercontent.com/43278267/169564506-90cf57ba-0e68-4a17-b442-f957e703e64c.jpg)

**Reply by fastweb-grupp** [reply]
its very important for photo quality and website magickstudio work cool, but ssh not cool
why???

**Reply by fmw42** [reply]
I explained the issue already. It has to do with the blending of pixel colors when reducing. If you do not want that, then use -sample rather than -resize.

**Reply by fastweb-grupp** [reply]
ok, but why function -resize work good in magickstudio? try and you see :)

**Comment by snibgo** [comment]
@fastweb-grupp: What is "SSH"?

JPG is a lossy format. This means that when you save an image as JPG, colours will change. You don't want colours to change, so don't use JPG. Use a lossless format such as JPG.

I encourage you to try my experiment: do the resizing at the command line, and in MagickStudio. Use "magick compare". What is the RMSE difference?

**Reply by fmw42** [reply]
Also be sure you use the same filter setting in MagickStudio as in command line.

**Reply by fastweb-grupp** [reply]
i use 1:1 and different result :)

**Reply by fmw42** [reply]
OK. Please provide the example input and output and your command line and we will test.

**Reply by fmw42** [reply]
Did you save as JPG or PNG or TIFF output?

**Reply by fmw42** [reply]
What version of Imagemagick (command line) were you using?

**Reply by fmw42** [reply]
Are you using MagickStudio from the Imagemagick web site or from some other mirror web site?

**Reply by fastweb-grupp** [reply]
ok, write againe :)

i use website https://magickstudio.imagemagick.org/scripts/MagickStudio.cgi i upload jpg file, make resize and get good result of image quality, but if use command line in unix /usr/local/bin/convert 36826.jpg -resize 360x240 -quality 100 36826-small.jpg i get different relult, some details of the image have dim colors.

**Reply by fmw42** [reply]
What settings did you use each step in MagickStudio?  Please provide your full size input image and the two full size output images so we can try to reproduce your results.
